### PR TITLE
Fix Project Submission bugs

### PR DESCRIPTION
**Pull request by rshipp**
### Description

Fix various bugs related to changing answers in a project after submitting.

* Correctly report 4xx and 5xx errors when changing Project answers.
* After submitting, the controller now creates and redirects to a new, unsubmitted submission
* After submitting, the React component updates the project submission id attribute stored in the wrapper div to the latest submission ID.
* Pull the project submission ID live from the div every time it's used in the project answers JS, instead of only once on page load, to support the above change.
* If you reload an old submission, instead of a plaintext error message, it now displays a more useful message with a link to access the page you probably expected to see.

### Motivation and Context

Task: https://app.asana.com/0/1174274412967132/1200410628872731/f

### Screenshots (if appropriate):
Screen Shot 2021-07-13 at 2 01 01 PM

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have added neccessary labels
- [ ] My PR title follows the convention in the guide
- [ ] I have assigned myself to the PR
- [ ] I have requested reviews from team members
- [ ] I have created a task for reviewers on PM tool
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/javascript/packs/project_answers.js [hunk 1]
```diff
@@ -160,23 +168,30 @@ export async function main() {
...
+                projectSubmitButton.toggleEnabled(false);
                 changeAutoSaveStatus('Some answers are still unsaved! Look for those that say "Failed to save answer." and click Retry.', 'autosave-status-error');
             }
-
-            // Logging.
-            honeySpan.addField('response.status', response.status, false);
         })
         .catch((error) => {
-            projectSubmitButtion.toggleEnabled(false);
+            projectSubmitButton.toggleEnabled(false);
```
Comment (rshipp) [inline app/javascript/packs/project_answers.js:194]: unrelated typo fix

File: app/javascript/components/Projects/ProjectSubmitButton.js [hunk 2]
```diff
@@ -56,9 +61,16 @@ class ProjectSubmitButton extends React.Component {
...
           'X-CSRF-Token': Rails.csrfToken(),
         },
+        redirect: 'follow',
       },
      )
     .then((response) => {
+      // Hack for half-React Projects.
+      if (response.redirected) {
+        const newProjectSubmissionId = response.url.match(PROJECT_SUBMISSION_ID_REGEX).groups.id;
+        this._updateProjectSubmissionId(newProjectSubmissionId);
```
Comment (sadleb) [inline app/javascript/components/Projects/ProjectSubmitButton.js:71]: Hard for me to tell if this can fail in any way and what would happen. Is there anyway this can throw? If so, what would the user experience be?

File: app/javascript/packs/project_answers.js [hunk 3]
```diff
@@ -160,23 +168,30 @@ export async function main() {
...
          )
         .then((response) => {
+            // Logging.
+            honeySpan.addField('response.status', response.status, false);
+
+            // Error handling.
+            if (!response.ok) {
+                // 4xx and 5xx responses should procede to the .catch below.
+                throw new Error(`HTTP Response ${response.status}`);
+                return;
```
Comment (sadleb) [inline app/javascript/packs/project_answers.js:178]: JS noob question, is this `return` ever hit?

**Review by sadleb — APPROVED** [review]
Questions, but nothing merge blocking.

File: app/javascript/packs/project_answers.js (hunk 3 above)
Comment (rshipp) [reply app/javascript/packs/project_answers.js:178]: I don't think so, but I also don't know js

File: app/javascript/components/Projects/ProjectSubmitButton.js (hunk 2 above)
Comment (rshipp) [reply app/javascript/components/Projects/ProjectSubmitButton.js:71]: If it breaks it should just act like the submission failed, which it probably did. most likely failure scenario is your session expired, the redirect was to the login page instead of the new submission, the regex doesn't match, `.groups` doesn't work, and there's some kind of null exception

File: app/javascript/packs/project_answers.js (hunk 3 above)
Comment (rshipp) [reply app/javascript/packs/project_answers.js:178]: https://stackoverflow.com/questions/26067190/do-i-need-to-return-after-throw-in-javascript
